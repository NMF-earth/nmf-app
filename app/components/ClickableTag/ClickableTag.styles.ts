import { StyleSheet } from "react-native";

import { Colors } from "style";

export default StyleSheet.create({
  default: {
    alignSelf: "center",
    paddingHorizontal: 10,
    paddingVertical: 4,
    borderRadius: 6,
    margin: 6,
    borderWidth: 1.5,
  },
  selected: {
    backgroundColor: Colors.primary10,
    borderColor: Colors.primary,
  },
  unselected: {
    backgroundColor: Colors.grey10,
    borderColor: Colors.black50,
  },
});
