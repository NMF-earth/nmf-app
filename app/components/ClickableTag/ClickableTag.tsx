import React, { ReactElement } from "react";
import { StyleProp, TouchableOpacity, ViewStyle } from "react-native";

import { Text } from "components";

import styles from "./ClickableTag.styles";

type Props = {
  text: string;
  isSelected: boolean;
  onPress: () => void;
};

const ClickableTag = ({ text, isSelected, onPress }: Props): ReactElement => {
  const containerStyle: StyleProp<ViewStyle> = [
    styles.default,
    isSelected ? styles.selected : styles.unselected,
  ];

  const textColor = { darkGray: !isSelected, green: isSelected };

  return (
    <TouchableOpacity style={containerStyle} onPress={onPress}>
      <Text.Secondary bold {...textColor}>
        {text}
      </Text.Secondary>
    </TouchableOpacity>
  );
};

export default ClickableTag;
