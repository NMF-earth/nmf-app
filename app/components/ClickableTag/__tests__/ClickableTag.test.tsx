import React from "react";
import { create } from "react-test-renderer";

import ClickableTag from "..";

describe("ClickableTag tests", () => {
  const mockOnPress = jest.fn();
  const props = {
    text: "Monday",
    isSelected: false,
    onPress: () => mockOnPress(),
  };

  it("should render correctly", () => {
    const element = create(<ClickableTag {...props} />).toJSON();

    expect(element).toMatchSnapshot();
  });

  // TODO: fix this
  // eslint-disable-next-line jest/no-commented-out-tests
  // it("given onPress should be called", () => {
  //   const element = create(<ClickableTag {...props} />).toJSON();
  //   element.props.onPress();

  //   expect(mockOnPress).toHaveBeenCalled();
  // });
});
