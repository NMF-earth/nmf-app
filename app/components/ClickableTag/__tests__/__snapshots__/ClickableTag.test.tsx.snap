// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickableTag tests should render correctly 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    [
      {
        "alignSelf": "center",
        "borderRadius": 6,
        "borderWidth": 1.5,
        "margin": 6,
        "paddingHorizontal": 10,
        "paddingVertical": 4,
      },
      {
        "backgroundColor": "#F0F0F0",
        "borderColor": "#878787",
      },
      undefined,
    ]
  }
>
  <Text.Secondary
    bold={true}
    darkGray={true}
    green={false}
  >
    Monday
  </Text.Secondary>
</TouchableOpacity>
`;
